tests/engines/xarray_engine/test_plugin.py

```diff
@@ -210,3 +210,15 @@ def test_indexing_array_nested_2D(self, simple_data_arrays):
         result = tiledb_data_array[[0, 2, 2], [1, 3]][[0, 0, 2], 1]
         expected = xarray_data_array[[0, 2, 2], [1, 3]][[0, 0, 2], 1]
         xr.testing.assert_allclose(result, expected)
+
+    def test_open_dataset_with_ctx(self):
+        tiledb_quickstart_dense = "tiledb://TileDB-Inc/quickstart_dense"
+        config = tiledb.Config()
+        config["rest.username"] = "a"
+        config["rest.password"] = "b"
+        ctx = tiledb.Ctx(config)
+        with pytest.raises(tiledb.libtiledb.TileDBError) as err:
+            xr.open_dataset(tiledb_quickstart_dense, engine="tiledb", ctx=ctx)
+        assert err.value.message.startswith(
+            "[TileDB::REST] Error: Error in libcurl GET operation"
+        )
```

Comment:
Testing that the context is correctly passed is gnarly. It would be best if we don't call into TileDB cloud for the test since that could fail for other unrelated reasons (e.g. no internet connectivity, wording in TileDB error changes).

After much searching, I can't find a way to easily get to the backend from the dataset after you open the dataset, so I don't immediately have a better solution.